packages/gatsby/src/redux/actions/public.js

```diff
@@ -357,15 +359,15 @@ ${reservedFields.map(f => `  * "${f}"`).join(`\n`)}
         page.component = trueComponentPath
       }
 
-      pageComponentCache[originalPageComponent] = page.component
+      pageComponentCache.set(originalPageComponent, page.component)
     }
   }
 
   let internalComponentName
   if (page.path === `/`) {
     internalComponentName = `ComponentIndex`
   } else {
-    internalComponentName = `Component${pascalCase(page.path)}`
```

Comment:
we were making this "pretty" for no good reason (it's not visible) and it just was burning CPU cycles.

Reply:
We actually never use it ourselves anymore. So I would suggest marking it as deprecated in https://github.com/gatsbyjs/gatsby/blob/172cf4d24ca6940839d0278db74e125cb7e8c682/packages/gatsby/src/schema/types/built-in-types.ts#L93
https://github.com/gatsbyjs/gatsby/blob/172cf4d24ca6940839d0278db74e125cb7e8c682/packages/gatsby/src/redux/types.ts#L28 add comment to remove this field completely in v4.

Reply:
Good idea, I'll do this in a follow-up PR.